pkg/vm/builtins/range_check.go

```diff
@@ -32,3 +40,34 @@ func (r *RangeCheck) InferValue(segment *memory.Segment, offset uint64) error {
 func (r *RangeCheck) String() string {
 	return RangeCheckName
 }
+
+func (r *RangeCheck) GetAllocatedSize(segmentUsedSize uint64, vmCurrentStep uint64) (uint64, error) {
+	allocatedInstances, err := GetAllocatedInstances(r.ratio, cellsPerRangeCheck, segmentUsedSize, instancesPerComponentRangeCheck, vmCurrentStep)
+	if err != nil {
+		return 0, err
+	}
+	return allocatedInstances * cellsPerRangeCheck, nil
+}
+
+func (r *RangeCheck) GetRangeCheckUsage(rangeCheckSegment *memory.Segment) (uint64, uint64) {
+	minVal, maxVal := ^uint64(0), uint64(0)
```

Comment:
minVal is max.uint64 and maxVal 0 ?

Reply:
Exactly! Those are just some corner values for that are quickly updated during the lookup